packages/authorization-updater/src/utils.ts

```diff
@@ -103,3 +116,81 @@ export const getPurposeVersionFromEvent = (
 
   return { purposeId: purpose.id, purposeVersion };
 };
+
+export const getClientFromEvent = (
+  msg: {
+    data: {
+      client?: ClientV2;
+    };
+  },
+  eventType: string
+): Client => {
+  if (!msg.data.client) {
+    throw missingKafkaMessageDataError("client", eventType);
+  }
+
+  return fromClientV2(msg.data.client);
+};
+
+export const apiClientKind = {
+  consumer: "CONSUMER",
+  api: "API",
+} as const;
+export const ApiClientKind = z.enum([
+  Object.values(apiClientKind)[0],
+  ...Object.values(apiClientKind).slice(1),
+]);
+export type ApiClientKind = z.infer<typeof ApiClientKind>;
+
+export const clientKindToApiClientKind = (kid: ClientKind): ApiClientKind =>
+  match<ClientKind, ApiClientKind>(kid)
+    .with(clientKind.consumer, () => "CONSUMER")
+    .with(clientKind.api, () => "API")
+    .exhaustive();
+
+export const apiKeyUse = {
+  sig: "SIG",
+  enc: "ENC",
+} as const;
+export const ApiKeyUse = z.enum([
+  Object.values(apiKeyUse)[0],
+  ...Object.values(apiKeyUse).slice(1),
+]);
+export type ApiKeyUse = z.infer<typeof ApiKeyUse>;
```

Comment:
This also should be in the `models.ts` file, taken from the generated api

```typescript
export type ApiKeyUse = z.infer<
  typeof api.schemas.KeyUse
>;

export const ApiClientComponent = api.schemas.KeyUse;
```

Reply:
Updated